### ISSUE TITLE
Fill in $remote_user

### ISSUE THREAD

**Issue by HazCod** [atomx/nginx-http-auth-digest#1]
Is it possible to fill in $remote_user with the username of the authenticated user?
This so I can correctly process my access logs.

Thank you

**Comment by ErikDubbelboer** [comment]
As you can see in the function that calculates `$remote_user` [ngx_http_variable_remote_user()](http://lxr.nginx.org/source/src/http/ngx_http_variables.c#1592) it uses [ngx_http_auth_basic_user()](http://lxr.nginx.org/source/src/http/ngx_http_core_module.c#2043) which is a core function which parses the authentication header but only supports `Basic`. A patch to nginx is required. I'll see what I can do when I have some time but I guess it will also take some time to be put in an nginx release.

**Comment by HazCod** [comment]
Ok, thank you.
I am using the workaround successfully for now.

**Comment by SubrataM3** [comment]
@erikdubbelboer 
Is this issue fixed ? it is still not working for me. 
I have compiled this module as dynamic module with nginx version: nginx/1.16.0 .

Please let me know.
OP's solution works for me though, but this should be inbuilt within module.

**Comment by erikdubbelboer** [comment]
@SubrataM3 I'm afraid this has not been fixed seeing as my patch to nginx was never accepted: http://mailman.nginx.org/pipermail/nginx-devel/2015-August/007209.html

**Comment by SubrataM3** [comment]
> @SubrataM3 I'm afraid this has not been fixed seeing as my patch to nginx was never accepted: http://mailman.nginx.org/pipermail/nginx-devel/2015-August/007209.html

Oh, that's bad. is it possible to bump that topic?

**Comment by erikdubbelboer** [comment]
A new attempt would have to be made. Seeing the last one was from 2015. But seeing as not a single person responded to me last time I'm not really encouraged to try again. I don't feel like the nginx maintainers are interested in patches like this.

**Comment by SubrataM3** [comment]
> A new attempt would have to be made. Seeing the last one was from 2015. But seeing as not a single person responded to me last time I'm not really encouraged to try again. I don't feel like the nginx maintainers are interested in patches like this.

Yep, I understand that, but since no one explicitly denied the patch i think you should try once more. digest auth is a basic feature and only your module supports it. That patch would be beneficial to many users. :-)

**Comment by erikdubbelboer** [comment]
I'm sorry but I really don't have the time for this at the moment. Could you maybe try to submit a patch to nginx?

**Comment by SubrataM3** [comment]
Hi, 
I can do that, but if they starts some discussion or ask question i won't be able to answer since i don't have any knowledge about nginx codebase .

**Comment by erikdubbelboer** [comment]
When that happens I'll help you answer any question.
